File: 2_currency/currency.cpp
#include <iostream>
#include <string>

using namespace std;

string lowercase(string s) {
    for (int i = 0; i < s.length(); i++) {
        s[i] = tolower(s[i]);
    };

    return s;
};

double eur2usd(double x) {
    // current conversion is 1 eur = 1.1829612 usd
    double r = 1.1829612;

    return x * r;
};

double usd2eur(double x) {
    // current conversion is 1 usd = 0.845379 eur
    double r = 0.845379;

    return x * r;
};

int main() {

    cout << "Which currency would you like to conver to (usd/eur)? ";
    string cur;
    cin >> cur;

    if (lowercase(cur) == "eur") {
        cout << "Enter amount of USD: ";
        double usd;
        cin >> usd;
        printf("%.2f", usd2eur(usd));
    } else if (lowercase(cur) == "usd") {
        cout << "Enter amount of EUR: ";
        double eur;
        cin >> eur;
        printf("%.2f", eur2usd(eur));
    } else {
        cout << "Unsupported currency";
    };
};

File: 2_currency/currency.jl
function usd2eur(x::Float64)::Float64
    # current conversion is 1 usd = 0.845379 eur
    r = 0.845379

    return x*r
end

function eur2usd(x::Float64)::Float64
    # current conversion is 1 eur = 1.1829612 usd
    r = 1.1829612

    return x*r
end

print("What currency would you like to convert to (usd/eur)? ")
cur = readline()
if lowercase(cur) == "usd"
    print("Enter an amount of EUR: ")
    eur = readline()
    eur = parse(Float64, eur)
    print(round(eur2usd(eur), digits=2))
elseif lowercase(cur) == "eur"
    print("Enter an amount of USD: ")
    usd = readline()
    usd = parse(Float64, usd)
    print(round(usd2eur(usd), digits=2))
else
    print("Unsupported currency")
end